### ISSUE TITLE
Implement hessians and newton's method for optimization in Python

### ISSUE THREAD

**Issue by suntzu86** [Yelp/MOE#57]
Currently python uses gradient descent for all optimization, even though hyperparameter optimization is _much_ faster with Newton's method.

implement hessian computation for:
covariance functions
log likelihood measures

and implement a NewtonOptimizer.

All this already exists in C++; so if you want to use it, call that instead.
